lib/config.js

```diff
@@ -63,20 +64,54 @@ const getAll = () => cloneDeep(config);
  * @param {*} value Value of a configuration property
  *
  * @returns {*} The value given.
+ *
+ * @deprecated since version 1.3
  */
 const set = (key, value) => {
-  config[key] = value;
+  // NOTE: we cannot use `log` function here as this module is a dependency of logger module.
+  console.warn('config.set is deprecated, please use config.init instead');
+
+  config = { ...config, [key]: value };
   return value;
 };
 
+let initialized = false;
+
 /**
  * Reset the configuration to defaults
+ *
+ * Reset the `initialized` boolean to false allowing config.init to be called again.
  */
 const reset = () => {
   config = cloneDeep(defaultConfig);
+  initialized = false;
```

Comment:
Question: What happens to our auth token if the config is reset? Do we trap ourselves in the same dilemma with standard and PII calls if, for whatever reason, the user sets `piiServerUrl`, resets, then doesn't set that prop again? 

This is likely an extreme edge case. Should we care? should we invalidate the auth token on reset? are there any other edge cases I'm not considering?